src/Sensors/IMU/IMU.cpp

```diff
@@ -1,51 +1,63 @@
 #include "IMU.h"
+#include <stdio.h>
 
 namespace mmfs
 {
+    void IMU::update()
+    {
+        read();
+    }
 
-    Quaternion IMU::getOrientationGlobal()
+    bool IMU::begin(bool useBiasCorrection)
     {
-        return orientationGlobal;
+        biasCorrectionMode = useBiasCorrection;
+        return init();
     }
 
-    Vector<3> IMU::getAccelerationGlobal()
+    Quaternion IMU::getOrientation()
     {
-        return accelerationGlobal;
+        return orientation;
     }
 
-    Quaternion IMU::getOrientationLocal()
+    Vector<3> IMU::getAngularVelocity()
     {
-        return orientationLocal;
+        return angularVelocity;
     }
 
-    Vector<3> IMU::getAccelerationLocal()
+    Vector<3> IMU::getMagField()
     {
-        return accelerationLocal;
+        return magField;
     }
 
-    Vector<3> IMU::getMagnetometerReading()
+    Vector<3> IMU::getAcceleration()
     {
-        return measuredMag;
+        return accelerationVec;
     }
 
-    Vector<3> IMU::getGyroReading()
+    Vector<3> IMU::getAccelerationGlobal()
     {
-        return measuredGyro;
+        Quaternion accelBody = Quaternion(0, accelerationVec.x(), accelerationVec.y(), accelerationVec.z());
+        Quaternion accelInterial = orientation * accelBody * orientation.conjugate();
+        return Vector<3>(accelInterial.x(), accelInterial.y(), accelInterial.z());
     }
 
-    Vector<3> IMU::getAccReading()
+    Vector<3> convertToEuler(const Quaternion &orientation)
```

Comment:
Is this a repeat of `Quaternion.h`'s method?

Reply:
yes. it was a relic of old code. good catch